# HG changeset patch
# User Colomban Wendling <ban@herbesfolles.org>
# Date 1487714930 -3600
# Node ID ae52e0c68190d423505624ae1d26e442e015d48f
# Parent  966f25e2241058f45500025b0a8e8ed4cae10ee7
Bug [#1901]. GTK: Fix double scrolling under X11

Really restrict smooth scrolling handling to Wayland.

diff -r 966f25e22410 -r ae52e0c68190 doc/ScintillaHistory.html
--- a/doc/ScintillaHistory.html	Sun Feb 19 14:13:46 2017 +1100
+++ b/doc/ScintillaHistory.html	Tue Feb 21 23:08:50 2017 +0100
@@ -520,6 +520,18 @@
       </li>
     </ul>
     <h3>
+       <a href="http://www.scintilla.org/scite374.zip">Release 3.7.4</a>
+    </h3>
+    <ul>
+	<li>
+	Released 19 February 2017.
+	</li>
+	<li>
+	Fix to prevent double scrolling on GTK+ with X11.
+	<a href="http://sourceforge.net/p/scintilla/bugs/1901/">Bug #1901</a>.
+	</li>
+    </ul>
+    <h3>
        <a href="http://www.scintilla.org/scite373.zip">Release 3.7.3</a>
     </h3>
     <ul>
diff -r 966f25e22410 -r ae52e0c68190 gtk/ScintillaGTK.cxx
--- a/gtk/ScintillaGTK.cxx	Sun Feb 19 14:13:46 2017 +1100
+++ b/gtk/ScintillaGTK.cxx	Tue Feb 21 23:08:50 2017 +0100
@@ -1801,7 +1801,7 @@
 			return FALSE;
 
 #if defined(GDK_WINDOWING_WAYLAND)
-		if (event->direction == GDK_SCROLL_SMOOTH) {
+		if (event->direction == GDK_SCROLL_SMOOTH && GDK_IS_WAYLAND_WINDOW(event->window)) {
 			const int smoothScrollFactor = 4;
 			sciThis->smoothScrollY += event->delta_y * smoothScrollFactor;
 			sciThis->smoothScrollX += event->delta_x * smoothScrollFactor;;
